app/models/notification_services/hipchat_service.rb

```diff
@@ -38,18 +42,34 @@ def url
     end
 
     def create_notification(problem)
-      url = app_problem_url problem.app, problem
-      message = <<-MSG.strip_heredoc
-        <strong>#{ERB::Util.html_escape problem.app.name}</strong> error in <strong>#{ERB::Util.html_escape problem.environment}</strong> at <strong>#{ERB::Util.html_escape problem.where}</strong> (<a href="#{url}">details</a>)<br>
-        &nbsp;&nbsp;#{ERB::Util.html_escape problem.message.to_s.truncate(100)}<br>
-        &nbsp;&nbsp;Times occurred: #{problem.notices_count}
-      MSG
+      format = self[:mentions].present? ? "text" : "html"
+      message = (format == "text") ? message_text(problem) : message_html(problem)
 
       options = { api_version: self[:service] }
       options[:server_url] = self[:service_url] if service_url.present?
 
       client = HipChat::Client.new(api_token, options)
-      client[room_id].send('Errbit', message, color: 'red', notify: true)
+      client[room_id].send('Errbit', message, color: 'red', notify: true, message_format: format)
+    end
+
+    private
+
+    def message_html(problem)
+      url = app_problem_url(problem.app, problem)
+      <<-MSG.strip_heredoc
+        <strong>#{ERB::Util.html_escape problem.app.name}</strong> error in <strong>#{ERB::Util.html_escape problem.environment}</strong> at <strong>#{ERB::Util.html_escape problem.where}</strong> (<a href="#{url}">details</a>)<br>
+        &nbsp;&nbsp;#{ERB::Util.html_escape problem.message.to_s.truncate(100)}<br>
+        &nbsp;&nbsp;Times occurred: #{problem.notices_count}
+      MSG
+    end
+
+    def message_text(problem)
+      url = app_problem_url(problem.app, problem)
+      <<-MSG
+#{self[:mentions]} #{problem.app.name} error in #{problem.environment}: #{problem.message.to_s.truncate(100)}
```

Comment:
If mentions is nil or blank (checked via mentions.present?), then the html format will be used.  html format uses the original behavior.  So users that have configured hipchat notifications without mentions will have the same behavior has before.  The only time you get the new behavior is when mentions is set.